azure: target must be a non-nil pointer with v4.1.0

Latest lego (v4.1.0) breaks azure dns provider.

Working fine with v4.0.1 and managed service identity

```
AZURE_RESOURCE_GROUP="xxxxxxxxxx" lego --email xxx@xxx.xxx --dns azure --domains "xxx.xxx.xxx" --accept-tos --server https://acme-staging-v02.api.letsencrypt.org/directory run
2020/11/16 15:49:59 No key found for xxx@xxx.xxx. Generating a P256 key.
2020/11/16 15:49:59 Saved key to /root/.lego/accounts/acme-staging-v02.api.letsencrypt.org/dxxx@xxx.xxx/keys/xxx@xxx.xxx
2020/11/16 15:50:00 [INFO] acme: Registering account for xxx@xxx.xxx
!!!! HEADS UP !!!!

Your account credentials have been saved in your Let's Encrypt
configuration directory at "/root/.lego/accounts".

You should make a secure backup of this folder now. This
configuration directory will also contain certificates and
private keys obtained from Let's Encrypt so making regular
backups of this folder is ideal.
2020/11/16 15:50:00 [INFO] [xxx.xxx.xxx] acme: Obtaining bundled SAN certificate
2020/11/16 15:50:01 [INFO] [xxx.xxx.xxx] AuthURL: https://acme-staging-v02.api.letsencrypt.org/acme/authz-v3/155728738
2020/11/16 15:50:01 [INFO] [xxx.xxx.xxx] acme: Could not find solver for: tls-alpn-01
2020/11/16 15:50:01 [INFO] [xxx.xxx.xxx] acme: Could not find solver for: http-01
2020/11/16 15:50:01 [INFO] [xxx.xxx.xxx] acme: use dns-01 solver
2020/11/16 15:50:01 [INFO] [xxx.xxx.xxx] acme: Preparing to solve DNS-01
panic: errors: target must be a non-nil pointer

goroutine 1 [running]:
errors.As(0x19453c0, 0xc000176000, 0x1474040, 0x0, 0x0)
        errors/wrap.go:84 +0x54f
github.com/go-acme/lego/v4/providers/dns/azure.(*DNSProvider).Present(0xc000540680, 0xc000026b80, 0x13, 0xc0000ad050, 0x2b, 0xc000428b40, 0x57, 0xc000362c88, 0x3)
        github.com/go-acme/lego/v4/providers/dns/azure/azure.go:186 +0x98f
github.com/go-acme/lego/v4/challenge/dns01.(*Challenge).PreSolve(0xc0003e9380, 0xc000362c90, 0x7, 0x0, 0xed74dc318, 0x0, 0xc000362c88, 0x3, 0xc000026b80, 0x13, ...)
        github.com/go-acme/lego/v4/challenge/dns01/dns_challenge.go:94 +0x2ce
github.com/go-acme/lego/v4/challenge/resolver.parallelSolve(0xc0003943b8, 0x1, 0x1, 0xc000402e40)
        github.com/go-acme/lego/v4/challenge/resolver/prober.go:135 +0x122
github.com/go-acme/lego/v4/challenge/resolver.(*Prober).Solve(0xc00000e058, 0xc000521f10, 0x1, 0x1, 0x1e, 0xc00057c180)
        github.com/go-acme/lego/v4/challenge/resolver/prober.go:84 +0x396
github.com/go-acme/lego/v4/certificate.(*Certifier).Obtain(0xc0003ac2d0, 0xc0000a5460, 0x1, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        github.com/go-acme/lego/v4/certificate/certificates.go:124 +0x2e2
github.com/go-acme/lego/v4/cmd.obtainCertificate(0xc000532160, 0xc00033e220, 0x178884b, 0x145, 0xc000751768)
        github.com/go-acme/lego/v4/cmd/cmd_run.go:171 +0x31d
github.com/go-acme/lego/v4/cmd.run(0xc000532160, 0xc0000b0200, 0x1)
        github.com/go-acme/lego/v4/cmd/cmd_run.go:88 +0xed
github.com/urfave/cli.HandleAction(0x1411380, 0x17b3788, 0xc000532160, 0xc000532160, 0x0)
        github.com/urfave/cli@v1.22.4/app.go:526 +0xfd
github.com/urfave/cli.Command.Run(0x1725a74, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0x177566f, 0x3a, 0x0, ...)
        github.com/urfave/cli@v1.22.4/command.go:174 +0x58e
github.com/urfave/cli.(*App).Run(0xc0000c81c0, 0xc00003a0b0, 0xb, 0xb, 0x0, 0x0)
        github.com/urfave/cli@v1.22.4/app.go:279 +0x7e8
main.main()
        command-line-arguments/main.go:42 +0x197
```

--- a/providers/dns/azure/azure.go
+++ b/providers/dns/azure/azure.go
@@ -183,7 +183,7 @@ func (d *DNSProvider) Present(domain, token, keyAuth string) error {
 	rset, err := rsc.Get(ctx, d.config.ResourceGroup, zone, relative, dns.TXT)
 	if err != nil {
 		var detailed *autorest.DetailedError
-		if !errors.As(err, detailed) || detailed.StatusCode != http.StatusNotFound {
+		if !errors.As(err, &detailed) || detailed.StatusCode != http.StatusNotFound {
 			return fmt.Errorf("azure: %w", err)
 		}
 	}

